src/transformStateWithClones.js

```diff
@@ -6,8 +6,35 @@
  *
  * @return {Object[]}
  */
+//
+//
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  const clonedState = { ...state };
+  const result = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(clonedState, action.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const key of action.keysToRemove) {
+          delete clonedState[key];
+        }
+        break;
+
+      default:
```

Comment:
Default case should log an error refering to improper input (read this https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/throw). In previous commit you correctly named this case "clear", however you can simplify your code->

case 'clear':
      clonedState = {};
      break;

Reply:
clear case = yours default
default case in switch:
`throw new Error('Unknown action type');`

PS: I don’t know why the system give me to do yours code review again